tests/mypy_test.py

```diff
@@ -118,6 +120,46 @@ def add_files(files, seen, root, name, args, exclude_list):
                         files.append(fn)
 
 
+class MypyDistConf(NamedTuple):
+    module_name: str
+    values: Dict
+
+# The configuration section in the metadata file looks like the following, with multiple module sections possible
+# [mypy-tests]
+# [mypy-tests.yaml]
+# module_name = "yaml"
+# [mypy-tests.yaml.values]
+# disallow_incomplete_defs = true
+# disallow_untyped_defs = true
+
+
+def add_configuration(configurations, seen_dist_configs, distribution):
+    if distribution in seen_dist_configs:
+        return
+
+    with open(os.path.join(THIRD_PARTY_NAMESPACE, distribution, "METADATA.toml")) as f:
+        data = dict(toml.loads(f.read()))
+
+    mypy_tests_conf = data.get("mypy-tests")
+    if not mypy_tests_conf:
+        return
+
+    assert isinstance(mypy_tests_conf, dict), "mypy-tests should be a section"
+    for section_name, mypy_section in mypy_tests_conf.items():
```

Comment:
The reason I put module_name in is because the module you use it for might not be a straight forward module or you might want to use a wildcard if you wanted the configuration to apply for several modules.
For example for protobuf, you can't put the section name as google.protobuf because the TOML heirarchy is dot based.